### DISCUSSION TITLE
Display matplotlib animations properly

### DISCUSSION THREAD

**Discussion by jburgy** [pyscript/pyscript#2253] [Proposals]
`pyscript.display` invokes the `savefig` method when the value to be displayed supports it.  That's not great for matplotlib animations as it only renders their first frame.  Looking at the [source](https://github.com/pyscript/pyscript/blob/main/core/src/stdlib/pyscript/display.py), I devised a simplistic work-around demonstrated [here](https://janburgy.pyscriptapps.com/attempt-at-matplotlib-animations/latest/).  How should `pyscript.display` change to support that feature?

**Comment by WebReflection** [comment]
Nice demo (it assumes `target` exists though, which is not a safe-assumption) ... my quick thoughts around this issue/question:

  1. `display` is a general-purpose (imho) last-resort utility to show something on the page when you don't know any better
  2. being general purpose, it would be weird to have tons of checks out of possibly unknown imported libraries out there, it will fail in a way or another with one library or another
  3. you do know what you are doing, and you don't need `current_target()` to accomplish that ... you just target that element created to show your animated demo and call it a day, the usage of `current_target()` out of a `tagName` check and whatnot, when you want to just inject whatever `to_jshtml()` returns to a DOM element, feels like way easier, and under your control, than us best-guessing your intent out of a more advanced use case that has specific requirements
  4. because of the previous point, your whole demo could be simplified as such: https://pyscript.com/@agiammarchi/attempt-at-matplotlib-animations-copy/latest?files=main.py

Accordingly, I don't think there's any action we should take, but happy to hear otherwise from @ntoll or others.

**Comment by jburgy** [comment]
@WebReflection makes a lot of sense, closing this proposal.

**Reply by WebReflection** [reply]
late note ... this:
```js
animation.replaceChildren(
    document.createRange().createContextualFragment(html)
)
```

could be just:
```js
animation.innerHTML = html
```

as I think you want to set the whole output anyway so there's no real-need to use 2 methods and invariant ways to result the same.

**Reply by WebReflection** [reply]
oops, never mind, that doesn't work as I think there is some part of the HTML that requires a contextual fragment indeed

**Comment by jburgy** [comment]
Hang on a minute, I just noticed that matplotlib animations actually implement `_repr_html_` so this might be a precedence issue.  Why does `savefig` appear _before_ `_repr_html_` in `_MIME_METHODS`?

**Comment by ntoll** [comment]
Hi folks,

Just catching up here... very cool demo. I think we should at the very least **document** how to do Matplotlib animations in PyScript along with a working demo. We can do this once we're all agreed on the best recipe to recommend. Likewise @jburgy - any feedback on code, or a PR to change order of calling (e.g. `savefig` and `_repr_html_`) would be welcome. I didn't write that code, but I don't think the original author was considering animations when they wrote it.
